src/sbvr-api/sbvr-utils.coffee

```diff
@@ -353,18 +358,59 @@ getHooks = do ->
 	_getHooks = (request) ->
 		if request.resourceName?
 			resourceName = resolveSynonym(request)
-		getMethodHooks(request.method, request.vocabulary, resourceName)
+		instantiateHooks(
+			getMethodHooks(
+				request.method
+				request.vocabulary
+				resourceName
+			)
+		)
 	_getHooks.clear = -> getMethodHooks.clear()
 	return _getHooks
 
-runHook = (hookName, args) ->
+class Hook
```

Comment:
This is a 100% optional btw, but it would be nice to separate these classes out to a `hooks.ts` or similar - if not then it'll happen at a later date along with the rest of the ts conversion